telegram_otherwise_test.go: flaky test

I've seen this more than once:

```
-- FAIL: TestUnitProcessallWithMixedResults (0.00s)
    telegram_otherwise_test.go:165: invalid TelegramWebFailure
FAIL
```

See:

- https://travis-ci.org/ooni/probe-engine/builds/620071781

- https://travis-ci.org/ooni/probe-engine/builds/620153895#L611

This is what I know for now:

1. cannot reproduce on macOS with `go test -race ./...` when telegram is manually modified to always run the Go version rather than the C++ version

2. does not happen on Linux with `CGO_ENABLED=0 go test -race ./...`

3. seems to happen on travis (four times in ~1 month)

Did see the failure again. It was like:

```
84.16s$ go test -coverprofile=coverage.cov -coverpkg=./... ./...
ok  	github.com/ooni/probe-engine	29.011s	coverage: 63.7% of statements in ./...
ok  	github.com/ooni/probe-engine/bouncer	0.444s	coverage: 3.4% of statements in ./...
?   	github.com/ooni/probe-engine/cmd/miniooni	[no test files]
ok  	github.com/ooni/probe-engine/collector	0.392s	coverage: 5.7% of statements in ./...
ok  	github.com/ooni/probe-engine/experiment	4.895s	coverage: 30.1% of statements in ./...
ok  	github.com/ooni/probe-engine/experiment/dash	0.008s	coverage: 7.1% of statements in ./...
ok  	github.com/ooni/probe-engine/experiment/example	6.058s	coverage: 28.1% of statements in ./...
ok  	github.com/ooni/probe-engine/experiment/fbmessenger	0.008s	coverage: 1.3% of statements in ./...
ok  	github.com/ooni/probe-engine/experiment/handler	0.007s	coverage: 1.4% of statements in ./...
ok  	github.com/ooni/probe-engine/experiment/hhfm	0.008s	coverage: 1.3% of statements in ./...
ok  	github.com/ooni/probe-engine/experiment/hirl	0.007s	coverage: 1.3% of statements in ./...
?   	github.com/ooni/probe-engine/experiment/httpheader	[no test files]
ok  	github.com/ooni/probe-engine/experiment/mkevent	0.008s	coverage: 3.3% of statements in ./...
ok  	github.com/ooni/probe-engine/experiment/mkhelper	0.007s	coverage: 2.4% of statements in ./...
ok  	github.com/ooni/probe-engine/experiment/mkrunner	0.009s	coverage: 3.1% of statements in ./...
ok  	github.com/ooni/probe-engine/experiment/ndt	0.020s	coverage: 1.3% of statements in ./...
ok  	github.com/ooni/probe-engine/experiment/ndt7	22.181s	coverage: 29.3% of statements in ./...
ok  	github.com/ooni/probe-engine/experiment/oodatamodel	0.011s	coverage: 4.7% of statements in ./...
ok  	github.com/ooni/probe-engine/experiment/psiphon	6.258s	coverage: 10.8% of statements in ./...
2019/12/07 16:20:08  info data usage: 0.0/0.0 down/up KiB
2019/12/07 16:20:08  info [ 8.3%] telegram: access http://149.154.171.5/: success
2019/12/07 16:20:09  info [16.7%] telegram: access http://149.154.171.5:443/: success
2019/12/07 16:20:09  info [25.0%] telegram: access http://149.154.167.51:443/: success
2019/12/07 16:20:09  info [33.3%] telegram: access http://149.154.167.91:443/: success
2019/12/07 16:20:09  info [41.7%] telegram: access http://web.telegram.org/: success
2019/12/07 16:20:10  info [50.0%] telegram: access http://149.154.175.100:443/: success
2019/12/07 16:20:10  info [58.3%] telegram: access http://149.154.167.91/: success
2019/12/07 16:20:11  info [66.7%] telegram: access http://149.154.175.100/: success
2019/12/07 16:20:11  info [75.0%] telegram: access https://web.telegram.org/: success
2019/12/07 16:20:11  info [83.3%] telegram: access http://149.154.175.50/: success
2019/12/07 16:20:12  info [91.7%] telegram: access http://149.154.175.50:443/: success
2019/12/07 16:20:12  info [100.0%] telegram: access http://149.154.167.51/: success
2019/12/07 16:20:12  info data usage: 12.2/3.9 down/up KiB
--- FAIL: TestUnitProcessallWithMixedResults (0.00s)
    telegram_otherwise_test.go:165: http_request_failed
    telegram_otherwise_test.go:167: invalid TelegramWebFailure
FAIL
coverage: 11.0% of statements in ./...
FAIL	github.com/ooni/probe-engine/experiment/telegram	4.687s
```

--- a/experiment/telegram/telegram_otherwise_test.go
+++ b/experiment/telegram/telegram_otherwise_test.go
@@ -151,7 +151,8 @@ func TestUnitProcessallWithMixedResults(t *testing.T) {
 		"https://web.telegram.org/": &urlMeasurements{
 			method: "GET",
 			results: &porcelain.HTTPDoResults{
-				Error: nil,
+				Error:      nil,
+				StatusCode: 200,
 			},
 		},
 	})

